include/zephyr/storage/flash_map.h

```diff
@@ -61,7 +62,7 @@ struct flash_area {
 	uint8_t fa_id;
 	uint16_t pad16;
 	/** Start offset from the beginning of the flash device */
-	off_t fa_off;
```

Comment:
@cfriedt The reason for esp32* mcuboot to fail is due to this particular change. primary_slot_partition and secondary_slot_partition define at https://github.com/zephyrproject-rtos/mcuboot/blob/main/boot/bootutil/src/loader.c#L2095 are getting 2x bigger, from 4k to 8k. Is that the expected?

Reply:
I get a similar increase when building for nrf52840dk/nrf52840 (@nordicjm)

v3.7.0-rc2:
```c
        Memory region         Used Size  Region Size  %age Used
           FLASH:       32830 B        48 KB     66.79%
             RAM:       23808 B       256 KB      9.08%
        IDT_LIST:          0 GB        32 KB      0.00%
```
This PR:
```
        Memory region         Used Size  Region Size  %age Used
           FLASH:       33406 B        48 KB     67.96%
             RAM:       25856 B       256 KB      9.86%
        IDT_LIST:          0 GB        32 KB      0.00%
        
 ```

Reply:
In case this is expected (RAM increase), I'll update esp32 mcuboot builds accordingly.